### PR TITLE
Fixed the scale of bcmod for PHP versions 7.2+

### PR DESCRIPTION
**Pull request by CodeDuck42**
Fixes the issue, that a previous bcscale screws with the precision of the bcmod function.

### CODE REVIEW

**Comment by jayawiperera** [comment]
@jschaedl Would it be possible to get this fix in ASAP? 

I've been having strange issues with many of my tests that deal with the IBAN validation. For some odd reason, with PHPUnit running a test suite / directory, the IBAN validation was failing in places where it should not. After much investigation, I came to realise that it had to do with the `bcscale` being set to 2, which caused `1.00` to be returned instead of `1`. 
I could not find where this is happening in my code, which leads me to believe it's something in configuration tucked away somewhere or a change happening due to another package. I was about to make a PR when I found this fix PR. 
Regardless of the cause, I believe this change will let us use the IBAN validator without issue and without causing other packages relying on `bcmath` to break.

File: src/Validator.php [hunk 1]
```diff
@@ -226,7 +226,11 @@ private function getNumericRepresentation($letterRepresentation)
     private function local_bcmod($operand, $modulus)
     {
         if (function_exists('bcmod')) {
-            return bcmod($operand, $modulus);
+            $oldScale = bcscale(0);
+            $mod = bcmod($operand, $modulus);
+            bcscale($oldScale);
```
Comment (xabbuh) [inline src/Validator.php]: it seems that this only works on PHP +7.3 as otherwise you would be passing `true` here: https://3v4l.org/tjCce

Comment (jayawiperera) [reply src/Validator.php]: Oof :(
This being the case, maybe we should use `ini_get` & `ini_set` for `bcmath.scale`?

Comment (CodeDuck42) [reply src/Validator.php]: Yeah, was a mistake on my side. I added a direct version check now.

File: src/Validator.php [hunk 2]
```diff
@@ -226,7 +226,9 @@ private function getNumericRepresentation($letterRepresentation)
     private function local_bcmod($operand, $modulus)
     {
         if (function_exists('bcmod')) {
-            return bcmod($operand, $modulus);
+            return version_compare(PHP_VERSION, '7.2') > 0
```
Comment (xabbuh) [inline src/Validator.php]: should be `PHP_VERSION_ID >= 70200` to allow the compiler to optimise this

File: src/Validator.php [hunk 3]
```diff
@@ -226,7 +226,9 @@ private function getNumericRepresentation($letterRepresentation)
     private function local_bcmod($operand, $modulus)
     {
         if (function_exists('bcmod')) {
-            return bcmod($operand, $modulus);
+            return PHP_VERSION_ID >= 70200
```
Comment (xabbuh) [inline src/Validator.php]: `>` right?

Comment (CodeDuck42) [reply src/Validator.php]: Its more inclusive this way 7.2.0 is also matched. Even though its unlikely that anybody uses it.

Comment (xabbuh) [reply src/Validator.php]: Sorry, what I meant was that this should only be executed for PHP +7.3 and not 7.2. But of course the correct comparison would have been `PHP_VERSION_ID >= 70300`.

Comment (CodeDuck42) [reply src/Validator.php]: bcscale was changed on 7.3+ but this fixes bcmod (7.2)

Comment (jschaedl) [reply src/Validator.php]: The `scale` argument of the `bcmod` function was introduced in `PHP 7.2.0`.
So `PHP_VERSION_ID >= 70200` looks good to me.

@xabbuh Or did I miss something?

Comment (xabbuh) [reply src/Validator.php]: Looks good to me. I missed that the patch was changed to not use `bcscale()` anymore.

**Comment by jschaedl** [comment]
Fixes #14

**Comment by jschaedl** [comment]
Thank you @CodeDuck42

**Pull request merged**